framework/include/materials/MaterialData.h

```diff
@@ -270,47 +278,25 @@ MaterialData::declarePropertyOlder(const std::string & prop_name)
   return *prop;
 }
 
-
 template<typename T>
 MaterialProperty<T> &
 MaterialData::getProperty(const std::string & name)
 {
-  unsigned int prop_id = _storage.getPropertyId(name);
-  resizeProps<T>(prop_id);
-
-  MaterialProperty<T> * prop = dynamic_cast<MaterialProperty<T> *>(_props[prop_id]);
-  if (prop != NULL)
-    return *prop;
-  else
-    mooseError("Material has no property named: " + name);
```

Comment:
I am still looking into/thinking about error checking.  If you look at the rest of the logic in MaterialData.h - this if clause was always true (i.e. this error was never thrown).